cohttp-eio/examples/client_tls.ml

```diff
@@ -0,0 +1,32 @@
+open Cohttp_eio
+
+let () =
+  Logs.set_reporter (Logs_fmt.reporter ());
+  Logs_threaded.enable ();
+  Logs.Src.set_level Cohttp_eio.src (Some Debug)
+
+let null_auth ?ip:_ ~host:_ _ =
+  Ok None (* Warning: use a real authenticator in your code! *)
+
+let https ~authenticator =
+  let tls_config = Tls.Config.client ~authenticator () in
+  fun uri raw ->
+    let host =
+      Uri.host uri
+      |> Option.map (fun x -> Domain_name.(host_exn (of_string_exn x)))
+    in
+    Tls_eio.client_of_flow ?host tls_config raw
+
+let () =
+  Eio_main.run @@ fun env ->
+  Mirage_crypto_rng_eio.run (module Mirage_crypto_rng.Fortuna) env @@ fun () ->
+  let client =
+    Client.make ~https:(Some (https ~authenticator:null_auth)) env#net
+  in
+  Eio.Switch.run @@ fun sw ->
+  let resp, body =
+    Client.get ~sw client (Uri.of_string "https://example.com")
```

Comment:
I am a bit sad with the amount of boilerplate we have to come up to do a simple `https` call. Could we somehow ship a default authenticator as well? The current API doesn't look very simple to use.

Reply:
We can't ship a default authenticator without adding extra dependencies to cohttp-eio. The plan is to have another package for that later (either cohttp-eio-tls, or maybe something conduit-like); this PR is just making that possible.

Reply:
I think we could merge this and release it as beta, so that people can test it out in the meantime and inform possible api refinements. But it would be nice to have `cohttp-eio-tls` before we release the new stable 6.0.0

Reply:
That sounds like a good plan. I'm already using that PR in a project of mine and it's working great 👍 Thanks for this!